src/DataStructures/SpinWeighted.hpp

```diff
@@ -26,60 +25,134 @@ struct SpinWeighted {
-/// spins are the same.
+/// spins are the same. Un-weighted quantities are assumed to be spin 0.
 template <typename T1, typename T2, int Spin>
 SPECTRE_ALWAYS_INLINE
     SpinWeighted<decltype(std::declval<T1>() + std::declval<T2>()), Spin>
     operator+(const SpinWeighted<T1, Spin>& lhs,
               const SpinWeighted<T2, Spin>& rhs) noexcept {
   return {lhs.data + rhs.data};
 }
+template <typename T>
+SPECTRE_ALWAYS_INLINE SpinWeighted<T, 0> operator+(
+    const SpinWeighted<T, 0>& lhs, const T& rhs) noexcept {
+  return {lhs.data + rhs};
+}
+template <typename T>
+SPECTRE_ALWAYS_INLINE SpinWeighted<T, 0> operator+(
+    const T& lhs, const SpinWeighted<T, 0>& rhs) noexcept {
+  return {lhs + rhs.data};
+}
+// @}
 
+// @{
 /// \brief Subtract two spin-weighted quantities if the types are compatible and
-/// spins are the same.
+/// spins are the same. Un-weighted quantities are assumed to be spin 0.
 template <typename T1, typename T2, int Spin>
 SPECTRE_ALWAYS_INLINE
-    SpinWeighted<decltype(std::declval<T1>() + std::declval<T2>()), Spin>
+    SpinWeighted<decltype(std::declval<T1>() - std::declval<T2>()), Spin>
     operator-(const SpinWeighted<T1, Spin>& lhs,
               const SpinWeighted<T2, Spin>& rhs) noexcept {
   return {lhs.data - rhs.data};
 }
+template <typename T>
+SPECTRE_ALWAYS_INLINE SpinWeighted<T, 0> operator-(
+    const SpinWeighted<T, 0>& lhs, const T& rhs) noexcept {
+  return {lhs.data - rhs};
+}
+template <typename T>
+SPECTRE_ALWAYS_INLINE SpinWeighted<T, 0> operator-(
+    const T& lhs, const SpinWeighted<T, 0>& rhs) noexcept {
+  return {lhs - rhs.data};
+}
+// @}
 
+// @{
 /// \brief Multiply two spin-weighted quantities if the types are compatible and
-/// add the spins.
+/// add the spins. Un-weighted quantities are assumed to be spin 0.
 template <typename T1, typename T2, int Spin1, int Spin2>
 SPECTRE_ALWAYS_INLINE SpinWeighted<
-    decltype(std::declval<T1>() + std::declval<T2>()), Spin1 + Spin2>
+    decltype(std::declval<T1>() * std::declval<T2>()), Spin1 + Spin2>
 operator*(const SpinWeighted<T1, Spin1>& lhs,
           const SpinWeighted<T2, Spin2>& rhs) noexcept {
   return {lhs.data * rhs.data};
 }
+template <typename T, int Spin>
+SPECTRE_ALWAYS_INLINE SpinWeighted<T, Spin> operator*(
+    const SpinWeighted<T, Spin>& lhs, const T& rhs) noexcept {
+  return {lhs.data * rhs};
+}
+template <typename T, int Spin>
+SPECTRE_ALWAYS_INLINE SpinWeighted<T, Spin> operator*(
+    const T& lhs, const SpinWeighted<T, Spin>& rhs) noexcept {
+  return {lhs * rhs.data};
+}
+// @}
 
+// @{
 /// \brief Divide two spin-weighted quantities if the types are compatible and
-/// subtract the spins.
+/// subtract the spins. Un-weighted quantities are assumed to be spin 0.
 template <typename T1, typename T2, int Spin1, int Spin2>
 SPECTRE_ALWAYS_INLINE SpinWeighted<
-    decltype(std::declval<T1>() + std::declval<T2>()), Spin1 - Spin2>
+    decltype(std::declval<T1>() / std::declval<T2>()), Spin1 - Spin2>
```

Comment:
this (and the similar fixes) should be in previous commit, or the last two commits should be squashed into one